File: Back/src/main/java/com/example/buensaborback/domain/entities/Factura.java
package com.example.buensaborback.domain.entities;

import com.example.buensaborback.domain.entities.enums.FormaPago;
import com.fasterxml.jackson.annotation.JsonFormat;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class Factura extends Base {

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "dd/MM/yyyy")
    private LocalDate fechaFacturacion;
    private Integer mpPaymentId;
    private Integer mpMerchantOrderId;
    private String mpPreferenceId;
    private String mpPaymentType;
    @Enumerated(EnumType.STRING)
    private FormaPago formaPago;
    private Double totalVenta;

    @OneToOne
    @JoinColumn(name = "pedido_id")
    private Pedido pedido;
}

File: Front/buenSaborFront/src/models/Factura.ts
import { PedidoCliente } from "./Pedido";

export default class Factura {
    id: number = 0;
    fechaFacturacion: string = "";
    mpPreferenceId?: string = "";
    montoDescuento: number = 0;
    formaPago: string = "";
    totalVenta: number = 0;
    totalCosto: number = 0;
    pedido: PedidoCliente = new PedidoCliente();
}